File: ClientApp/app/models/weekentry.ts
export interface Category {
    id: number;
    name: string;
    number: number;
}

export interface Week {
    id: number; 
    quarter: number; 
    number: number; 
    year: number; 
    entries: Array<any>;
}

export interface WeekEntry {
    id: number;
    category: Category;
    week: Week;
    mon: number;
    tue: number;
    wed: number;
    thurs: number;
    fri: number;
    lastUpdated: string;
}

export interface SaveWeekEntry {
    id: number;
    categoryId: number;
    weekId: number;
    mon: number;
    tue: number;
    wed: number;
    thurs: number;
    fri: number;
}

File: Core/Models/WeekEntry.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.ComponentModel.DataAnnotations;
using ACR2.Core.Models;

namespace ACR2.Models
{
    public class WeekEntry
    {
        public int Id { get; set; }
        [Required]
        public int CategoryId { get; set; }
        public Category Category { get; set; }
        [Required]
        public int Mon { get; set; }
        [Required]
        public int Tue { get; set; } 
        [Required]
        public int Wed { get; set; }
        [Required]
        public int Thurs { get; set; }
        [Required]
        public int Fri { get; set; }
        [Required]
        public int WeekId { get; set; }
        public Week Week { get; set; }
        public string Auth0Id { get; set; }
        [Required]
        public DateTime LastUpdated { get; set; }
        public ICollection<Photo> Photos { get; set; }

        public WeekEntry()
        {
            Photos = new Collection<Photo>();
        }
    }
}